engine/api/src/main/scala/pl/touk/nussknacker/engine/api/ModelConfigLoader.scala

```diff
@@ -0,0 +1,31 @@
+package pl.touk.nussknacker.engine.api
+
+import com.typesafe.config.Config
+
+/**
+  * Loads model configs. Base config comes from processTypes.{typeName}.modelConfig.
+  *
+  * Configs are resolved once when loading model data. Implement this trait when you want to add dynamic configuration
+  * that is fetched from external source, for example OpenAPI specification and you would like this configuration to
+  * be immutable for deployed processes.
+  *
+  * This trait is optional to implement. Default implementations complements base config with configuration
+  * available in JAR's resources.
+  */
+trait ModelConfigLoader extends Serializable {
+
+  /**
+    *  Resolves full config used inside [[pl.touk.nussknacker.engine.api.process.ProcessConfigCreator]]. Invoked both
+    *  in NK and running process.
+    */
+  def resolveFullConfig(configPassedInExecution: Config, classLoader: ClassLoader): Config
```

Comment:
I think we should provide default implementation out of the box here. I think it should be very conscious decision, that defaultModelConfig etc. will not be used.